pkg/services/validator/oci.go

```diff
@@ -361,3 +361,80 @@ func readArtifactRef(r *plug.OciRegistryRule, a *plug.Artifact, idx int) error {
 	}
 	return nil
 }
+
+func readOciSecret(secret *components.Secret) error {
+	var err error
+	if secret.Name == "" {
+		secret.Name, err = prompts.ReadK8sName("Secret Name", "", false)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.InfoCLI("Reconfiguring secret: %s", secret.Name)
+	}
+
+	// reconfigure and/or add basic auth
+	if secret.BasicAuth == nil {
+		secret.BasicAuth = &components.BasicAuth{}
+	}
+	addBasicAuth, err := prompts.ReadBool("Add basic auth to this secret", false)
+	if err != nil {
+		return err
+	}
+	if addBasicAuth {
+		secret.BasicAuth.Username, secret.BasicAuth.Password, err = prompts.ReadBasicCreds(
+			"Username", "Password", secret.BasicAuth.Username, secret.BasicAuth.Password, false, false,
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		secret.BasicAuth = nil
+	}
+
+	// reconfigure and/or add env vars
+	if secret.Data == nil {
+		secret.Data = make(map[string]string)
+	}
+	for k, v := range secret.Data {
+		secret.Data[k], secret.Data[v], err = readKeyValue(k, v)
```

Comment:
if the secret.Data map is keys -> vals, shouldnt this instead be:
```suggestion
        key, data, err := readKeyValue(k,v)
        if err != nil {
            return err
        }
        secret.Data[key] = val
```
As is, it looks like data is being stored like this:
k -> key from readKeyValue()
v -> val from readKeyValue()

Reply:
Good catch! Definitely my bad!